src/Symfony/Bridge/Doctrine/Form/Type/EntityType.php

```diff
@@ -27,11 +33,60 @@ class EntityType extends DoctrineType
      */
     public function getLoader(ObjectManager $manager, $queryBuilder, $class)
     {
-        return new ORMQueryBuilderLoader(
-            $queryBuilder,
-            $manager,
-            $class
-        );
+        $queryBuilderHash = null;
+
+        if ($queryBuilder instanceof QueryBuilder) {
+            $queryBuilderHash = $this->getQueryBuilderHash($queryBuilder);
+        }
+
+        if (null === $queryBuilderHash) {
```

Comment:
the `queryBuilderhash` must not be null if `$queryBuilder instanceof QueryBuilder` i dont like the structure of the if conditions here.